apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -334,14 +383,19 @@ public SpannerConnectionStringBuilder(string connectionString, ChannelCredential
         public SpannerConnectionStringBuilder(string connectionString, ChannelCredentials credentials, SessionPoolManager sessionPoolManager)
         {
             ConnectionString = GaxPreconditions.CheckNotNull(connectionString, nameof(connectionString));
-            CredentialOverride = credentials;
+            // If SPANNER_EMULATOR_HOST env var is set, ignore any provided credentials and use insecure credentials.
+            ParseEmulatorHostAndPort();
+            CredentialOverride = string.IsNullOrEmpty(_emulatorHost) ? credentials : ChannelCredentials.Insecure;
             SessionPoolManager = GaxPreconditions.CheckNotNull(sessionPoolManager, nameof(sessionPoolManager));
         }
 
         /// <summary>
         /// Creates a new <see cref="SpannerConnectionStringBuilder"/>.
         /// </summary>
-        public SpannerConnectionStringBuilder() { }
+        public SpannerConnectionStringBuilder() : this("")
```

Comment:
null, not "". In general avoid the empty string as a signal for an unknown value. It carries the same problems as a 0 for an unknown int (thankfully we now have int?).

Reply:
No longer making this change.